SmartDeviceLinkTests/DevAPISpecs/SDLTextAndGraphicUpdateOperationSpec.m

```diff
@@ -0,0 +1,929 @@
+                    expect(testOp.currentScreenData.mediaTrack).to(beEmpty());
+                    expect(testOp.currentScreenData.graphic).to(beNil());
+                    expect(testOp.currentScreenData.secondaryGraphic).to(beNil());
+                    expect(testOp.currentScreenData.metadataTags.mainField1).toNot(beNil());
+                    expect(testOp.currentScreenData.metadataTags.mainField2).toNot(beNil());
+                    expect(testOp.currentScreenData.metadataTags.mainField3).toNot(beNil());
+                    expect(testOp.currentScreenData.metadataTags.mainField4).to(beNil());
+                });
+            });
+
+            context(@"when sending four lines of text", ^{
+                beforeEach(^{
+                    updatedState = [[SDLTextAndGraphicState alloc] init];
+                    updatedState.textField1 = field1String;
+                    updatedState.textField2 = field2String;
+                    updatedState.textField3 = field3String;
+                    updatedState.textField4 = field4String;
+
+                    testOp = [[SDLTextAndGraphicUpdateOperation alloc] initWithConnectionManager:testConnectionManager fileManager:mockFileManager currentCapabilities:windowCapability currentScreenData:emptyCurrentDataShow newState:updatedState currentScreenDataUpdatedHandler:nil updateCompletionHandler:nil];
+                    [testOp start];
+
+                    [testConnectionManager respondToLastRequestWithResponse:successShowResponse];
+                });
+
+                it(@"should only send one line of text", ^{
+                    expect(testOp.isFinished).to(beTrue());
+                    expect(testOp.currentScreenData.mainField1).to(equal([NSString stringWithFormat:@"%@", field1String]));
+                    expect(testOp.currentScreenData.mainField2).to(equal([NSString stringWithFormat:@"%@", field2String]));
+                    expect(testOp.currentScreenData.mainField3).to(equal([NSString stringWithFormat:@"%@ - %@", field3String, field4String]));
+                    expect(testOp.currentScreenData.mainField4).to(beEmpty());
+                    expect(testOp.currentScreenData.templateTitle).to(beEmpty());
+                    expect(testOp.currentScreenData.alignment).to(beNil());
+                    expect(testOp.currentScreenData.mediaTrack).to(beEmpty());
+                    expect(testOp.currentScreenData.graphic).to(beNil());
+                    expect(testOp.currentScreenData.secondaryGraphic).to(beNil());
+                    expect(testOp.currentScreenData.metadataTags.mainField1).toNot(beNil());
+                    expect(testOp.currentScreenData.metadataTags.mainField2).toNot(beNil());
+                    expect(testOp.currentScreenData.metadataTags.mainField3).toNot(beNil());
+                    expect(testOp.currentScreenData.metadataTags.mainField4).to(beNil());
+                });
+            });
+        });
+
+        // with four lines available
```

Comment:
The reason I put all of these comments is because I found that the best way to navigate a spec is to fold the blocks in Xcode. But, then you can't see the name of the block. So I started putting comments so that I can fold the blocks and still see what the block is. If you agree that that's a good method of navigating specs, I think we should start doing that as much as we can, at least in complex specs.